src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -189,3 +194,74 @@ trait IterablePolyTransforms[+A, +C[A]] extends Any {
   def zip[B](xs: IterableOnce[B]): C[(A @uncheckedVariance, B)] = fromIterable(View.Zip(coll, xs))
   // sound bcs of VarianceNote
 }
+
+/** Transforms over iterables that can return collections of different element types for which an
+  * implicit evidence is required.
+  */
```

Comment:
Should explain in comment with examples why we need both `C` and `CC`. It looked non-senscial at first tome, and I started to understand only when I looked at instantiations of the class. (of course if we found a way to merge `C` and `CC` that would be even better).